pkg/bff/db/organizations.go

```diff
@@ -1,118 +1,36 @@
 package db
 
 import (
-	"context"
-	"time"
-
 	"github.com/google/uuid"
 	"github.com/trisacrypto/directory/pkg/bff/db/models/v1"
-	"google.golang.org/protobuf/proto"
-)
-
-const (
-	NamespaceOrganizations = "organizations"
 )
 
-// The Organizations collection exposes methods for interacting with Organization
-// records in the database. Organizations are given UUIDs when created, and those uuids
-// are also used as keys in the database for quick Gets by ID.
-//
-// Organizations implements the Collection interface
-type Organizations struct {
-	db        *DB
-	namespace string
-}
-
-// Ensure that Organizations implements the Collection interface.
-var _ Collection = &Organizations{}
-
-// Organizations constructs the collection type for db interactions with the namespace.
-// This method is intended to be used with chaining, e.g. db.Organizations().Get(id). To
-// reduce the number of allocations a singleton is used. Method calls to the collection
-// are thread-safe.
-func (db *DB) Organizations() *Organizations {
-	db.makeOrganizations.Do(func() {
-		db.organizations = &Organizations{
-			db:        db,
-			namespace: NamespaceOrganizations,
-		}
-	})
-	return db.organizations
-}
-
-// Create an organization and get back the organization stub.
-func (o *Organizations) Create(ctx context.Context) (org *models.Organization, err error) {
-	// Create an empty organization
-	ts := time.Now().Format(time.RFC3339Nano)
-	uu := uuid.New()
-	org = &models.Organization{
-		Id:       uu.String(),
-		Created:  ts,
-		Modified: ts,
-	}
-
-	var value []byte
-	if value, err = proto.Marshal(org); err != nil {
-		return nil, err
-	}
-
-	if err = o.db.Put(ctx, uu[:], value, o.namespace); err != nil {
-		return nil, err
-	}
-
-	return org, nil
+// Create an organization and get back the organization stub with an assigned ID.
+func (store *DB) CreateOrganization() (*models.Organization, error) {
+	return store.db.CreateOrganization()
```

Comment:
Is it necessary to have these indirects or the `db` module at all? Shouldn't we just delete the db module and use the store directly from the BFF?

Reply:
I agree that the db module is a bit unnecessary at this point, and we can just move the high level announcement methods to the BFF server struct instead.